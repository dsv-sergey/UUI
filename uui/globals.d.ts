declare module '*.scss' {
    const all: { [selector: string]: string };
    export = all;
}

declare module '*.svg' {
    export const id: string;
    export const url: string;
    export const viewBox: string;
}

declare module "jsurl";
declare module "query-string";
