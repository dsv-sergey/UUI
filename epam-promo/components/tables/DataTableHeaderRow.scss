@use  '../../assets/styles/index' as *;

.root {
    background: $gray5;
    border-bottom: 1px solid $gray30;
}

.header-cell {
    background: none;
}

.config-icon {
    position: absolute;
    display: flex;
    right: 24px;
    height: 100%;
}
