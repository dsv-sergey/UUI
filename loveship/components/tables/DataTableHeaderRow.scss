@use '../../assets/styles/scss/index'  as *;

.root {
    background: $night50;
    border-bottom: 1px solid $night300;
}

.header-cell {
    background: none;
}

.config-icon {
    position: absolute;
    right: 24px;
    height: 100%;
}
