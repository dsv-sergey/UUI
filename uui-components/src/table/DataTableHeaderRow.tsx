import * as React from 'react';
import {uuiContextTypes, DataSourceState, SortDirection, DataColumnProps, DataTableHeaderRowProps, Lens, DataTableHeaderCellProps,
    getColumnsConfig, DropParams, getOrderBetween} from '@epam/uui';
import { DataTableRowContainer } from './DataTableRowContainer';

const uuiDataTableHeaderRow = {
    uuiTableHeaderRow: 'uui-table-header-row',
};

export class DataTableHeaderRow<T> extends React.Component<DataTableHeaderRowProps<T, any>, {}> {
    static contextTypes = uuiContextTypes;

    lens = Lens.onEditableComponent<DataSourceState>(this);
    sortLens = this.lens.prop('sorting');
    filterLens = this.lens.prop('filter');

    onCellDrop = (params: DropParams<DataColumnProps<any>, DataColumnProps<any>>, index: number) => {
        const columnsConfig = getColumnsConfig(this.props.columns, this.props.value.columnsConfig);

        const dstColumnConfig = columnsConfig[params.dstData.key];
        const srcColumnConfig = columnsConfig[params.srcData.key];

        const prevColumnOrder = this.props.columns[index - 1] ? columnsConfig[this.props.columns[index - 1].key].order : null;
        const nextColumnOrder = this.props.columns[index + 1] ? columnsConfig[this.props.columns[index + 1].key].order : null;

        if (params.position === 'left') {
            const newOrder = getOrderBetween(prevColumnOrder, dstColumnConfig.order);
            columnsConfig[params.srcData.key] = { ...srcColumnConfig, order: newOrder };
        } else if (params.position === 'right') {
            const newOrder = getOrderBetween(dstColumnConfig.order, nextColumnOrder);
            columnsConfig[params.srcData.key] = { ...srcColumnConfig, order: newOrder };
        }

        this.props.onValueChange({ ...this.props.value, columnsConfig });
    }

    renderCell = (column: DataColumnProps<any, any>, idx: number) => {
        const sorting = this.sortLens.index(0).default({ field: null, direction: 'asc' }).get();

        const cellProps: DataTableHeaderCellProps<T, any> = {
            column,
            value: this.props.value,
            onValueChange: this.props.onValueChange,
            selectAll: this.props.selectAll,
            isFirstColumn: idx === 0,
            isLastColumn: idx === this.props.columns.length - 1,
            isFilterActive: column.isFilterActive && column.isFilterActive(this.filterLens.default({}).get(), column),
            sortDirection: sorting.field === column.key ? (sorting.direction || 'asc') : null,
            allowColumnsReordering: this.props.allowColumnsReordering,
            allowColumnsResizing: this.props.allowColumnsResizing,
            onSort: (dir: SortDirection) => this.props.onValueChange({
                ...this.props.value,
                sorting: (sorting.field !== column.key || sorting.direction !== dir) ? [{ field: column.key, direction: dir }] : [],
            }),
            onDrop: (params) => this.onCellDrop(params, idx),
            renderFilter: () => column.renderFilter(this.filterLens),
        };

        return this.props.renderCell(cellProps);
    }

    render() {
        return (
            <DataTableRowContainer
                cx={ [this.props.cx, uuiDataTableHeaderRow.uuiTableHeaderRow] }
                scrollManager={ this.props.scrollManager }
                columns={ this.props.columns }
                renderCell={ this.renderCell }
                renderConfigButton={ this.props.onConfigButtonClick && this.props.renderConfigButton }
            />
        );
    }
}
