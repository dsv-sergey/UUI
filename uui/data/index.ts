export * from './Stateful';
export * from './querying';
export * from './processing';
export * from './lenses';
export * from './forms';
