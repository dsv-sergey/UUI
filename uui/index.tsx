export * from './constants';
export * from './helpers';
export * from './services';
export * from './types';
export * from './data';
export * from './i18n';
export * from './testData';