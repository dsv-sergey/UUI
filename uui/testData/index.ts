import {dataSource} from "./dataSource";
import {dataColumns} from "./dataColumns";
import {dataSourceView} from "./dataSourceView";

export const testData = {
    dataSource,
    dataColumns,
    dataSourceView,
};