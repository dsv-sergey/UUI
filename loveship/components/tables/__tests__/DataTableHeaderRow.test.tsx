import React from "react";
import renderer from "react-test-renderer";
import {testData} from "@epam/uui";
import {DataTableHeaderRow} from "../DataTableHeaderRow";

jest.mock("react-dom", () => ({
    findDOMNode: jest.fn(),
}));

describe("DataTableHeaderRow", () => {
    it("should render with default props", () => {
        const tree = renderer
            .create(<DataTableHeaderRow value={ null }
                                        onValueChange={ jest.fn() }
                                        columns={ testData.dataColumns }/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should be rendered correctly with extra props", () => {
        const tree = renderer
            .create(<DataTableHeaderRow
                value={ {
                    topIndex: 0,
                    visibleCount: 10,
                } }
                onValueChange={ jest.fn() }
                columns={ testData.dataColumns }
                size='24'
                textCase='upper'
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});