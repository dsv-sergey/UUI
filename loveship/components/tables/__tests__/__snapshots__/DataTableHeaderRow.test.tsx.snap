// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableHeaderRow should be rendered correctly with extra props 1`] = `
<div
  className="container uui-table-row-container root uui-table-header-row container align-items-center"
>
  <div
    className="scrollableColumnsWrapper"
    style={
      Object {
        "flexGrow": 100,
        "flexShrink": 1,
        "minWidth": 0,
        "width": 0,
      }
    }
  >
    <div
      className="scrollableColumnsContainer"
    >
      <div
        className="container align-items-center"
      >
        <div
          className="flexCell -clickable cell size-24 padding-left-24"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "96px",
              "flexGrow": 0,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <div
            className="tooltipWrapper"
            onClick={null}
          >
            <div
              className="iconCell align-undefined"
            >
              <div
                className="text font-sans color-night700 line-height-24 font-size-14 v-padding-6 caption overflowedCell upperCase container"
              >
                ID
              </div>
              <div
                className="container uui-button-box uui-enabled root size-24 color-night400 font-sans-semibold uui-has-left-icon uui-no-right-icon icon sortIcon"
                href={null}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="container uui-icon uui-enabled"
                >
                  <svg
                    className=""
                    height={24}
                    viewBox="0 0 12 24"
                    width={12}
                  >
                    <use
                      xlinkHref="#sort.svg"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="resizeMark"
            onMouseDown={[Function]}
          />
        </div>
        <div
          className="flexCell -clickable cell size-24"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "96px",
              "flexGrow": 0,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <div
            className="tooltipWrapper"
            onClick={null}
          >
            <div
              className="iconCell align-undefined"
            >
              <div
                className="text font-sans color-night700 line-height-24 font-size-14 v-padding-6 caption overflowedCell upperCase container"
              >
                Level
              </div>
              <div
                className="container uui-button-box uui-enabled root size-24 color-night400 font-sans-semibold uui-has-left-icon uui-no-right-icon icon sortIcon"
                href={null}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="container uui-icon uui-enabled"
                >
                  <svg
                    className=""
                    height={24}
                    viewBox="0 0 12 24"
                    width={12}
                  >
                    <use
                      xlinkHref="#sort.svg"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="resizeMark"
            onMouseDown={[Function]}
          />
        </div>
        <div
          className="flexCell -clickable cell size-24 padding-right-24"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "96px",
              "flexGrow": 0,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <div
            className="tooltipWrapper"
            onClick={null}
          >
            <div
              className="iconCell align-undefined"
            >
              <div
                className="text font-sans color-night700 line-height-24 font-size-14 v-padding-6 caption overflowedCell upperCase container"
              >
                DATE
              </div>
              <div
                className="container uui-button-box uui-enabled root size-24 color-night400 font-sans-semibold uui-has-left-icon uui-no-right-icon icon sortIcon"
                href={null}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="container uui-icon uui-enabled"
                >
                  <svg
                    className=""
                    height={24}
                    viewBox="0 0 12 24"
                    width={12}
                  >
                    <use
                      xlinkHref="#sort.svg"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="resizeMark"
            onMouseDown={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="scrollShadowLeft"
    />
    <div
      className="scrollShadowRight"
    />
  </div>
</div>
`;

exports[`DataTableHeaderRow should render with default props 1`] = `
<div
  className="container uui-table-row-container root uui-table-header-row container align-items-center"
>
  <div
    className="scrollableColumnsWrapper"
    style={
      Object {
        "flexGrow": 100,
        "flexShrink": 1,
        "minWidth": 0,
        "width": 0,
      }
    }
  >
    <div
      className="scrollableColumnsContainer"
    >
      <div
        className="container align-items-center"
      >
        <div
          className="flexCell -clickable cell size-36 padding-left-24"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "96px",
              "flexGrow": 0,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <div
            className="tooltipWrapper"
            onClick={null}
          >
            <div
              className="iconCell align-undefined"
            >
              <div
                className="text font-sans color-night700 line-height-24 font-size-14 v-padding-6 caption overflowedCell font-size-14 container"
              >
                ID
              </div>
              <div
                className="container uui-button-box uui-enabled root size-24 color-night400 font-sans-semibold uui-has-left-icon uui-no-right-icon icon sortIcon"
                href={null}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="container uui-icon uui-enabled"
                >
                  <svg
                    className=""
                    height={24}
                    viewBox="0 0 12 24"
                    width={12}
                  >
                    <use
                      xlinkHref="#sort.svg"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="resizeMark"
            onMouseDown={[Function]}
          />
        </div>
        <div
          className="flexCell -clickable cell size-36"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "96px",
              "flexGrow": 0,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <div
            className="tooltipWrapper"
            onClick={null}
          >
            <div
              className="iconCell align-undefined"
            >
              <div
                className="text font-sans color-night700 line-height-24 font-size-14 v-padding-6 caption overflowedCell font-size-14 container"
              >
                Level
              </div>
              <div
                className="container uui-button-box uui-enabled root size-24 color-night400 font-sans-semibold uui-has-left-icon uui-no-right-icon icon sortIcon"
                href={null}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="container uui-icon uui-enabled"
                >
                  <svg
                    className=""
                    height={24}
                    viewBox="0 0 12 24"
                    width={12}
                  >
                    <use
                      xlinkHref="#sort.svg"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="resizeMark"
            onMouseDown={[Function]}
          />
        </div>
        <div
          className="flexCell -clickable cell size-36 padding-right-24"
          onClick={[Function]}
          style={
            Object {
              "alignSelf": undefined,
              "flexBasis": "96px",
              "flexGrow": 0,
              "flexShrink": 0,
              "minWidth": 0,
              "textAlign": undefined,
            }
          }
        >
          <div
            className="tooltipWrapper"
            onClick={null}
          >
            <div
              className="iconCell align-undefined"
            >
              <div
                className="text font-sans color-night700 line-height-24 font-size-14 v-padding-6 caption overflowedCell font-size-14 container"
              >
                DATE
              </div>
              <div
                className="container uui-button-box uui-enabled root size-24 color-night400 font-sans-semibold uui-has-left-icon uui-no-right-icon icon sortIcon"
                href={null}
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
              >
                <div
                  className="container uui-icon uui-enabled"
                >
                  <svg
                    className=""
                    height={24}
                    viewBox="0 0 12 24"
                    width={12}
                  >
                    <use
                      xlinkHref="#sort.svg"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="resizeMark"
            onMouseDown={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="scrollShadowLeft"
    />
    <div
      className="scrollShadowRight"
    />
  </div>
</div>
`;
